evennia/contrib/game_systems/achievements/README.md

```diff
@@ -0,0 +1,119 @@
+- **tracking_type** (str): *Default value if unset: `"sum"`* There are two valid tracking types: "sum" (which is the default) and "separate". `"sum"` will increment a single counter every time any of the tracked items match. `"separate"` will have a counter for each individual item in the tracked items. (This is really only useful when `"tracking"` is a list.)
+- **count** (int): *Default value if unset: 1* The number of tallies this achievement's requirements need to build up in order to complete the achievement. e.g. the previous example of killing rats would have a `"count"` value of `10`. For achievements using the "separate" tracking type, *each* item being tracked must tally up to this number to be completed
+- **prereqs** (str or list): The *keys* of any achievements which must be completed before this achievement can start tracking progress. An achievement's key is the variable name it's assigned to in your achievement module.
+
+You can add any additional keys to your achievement dicts that you want, and they'll be included with all the rest of the achievement data when using the contrib's functions. This could be useful if you want to add extra metadata to your achievements for your own features.
+
+#### Examples
+
+A simple achievement which you can get just for logging in the first time. This achievement has no prerequisites and it only needs to be fulfilled once to complete, so it doesn't need to define most of the keys.
+```python
+# This achievement has the unique key of "FIRST_LOGIN_ACHIEVE"
+FIRST_LOGIN_ACHIEVE = {
+    "name": "Welcome!", # the searchable, player-friendly display name
+    "desc": "We're glad to have you here.", # the longer description
+    "category": "login", # the type of action this tracks
+    "tracking": "first", # the specific login action
+}
+```
+
+An achievement for killing 10 rats, and another for killing 10 *dire* rats which requires the "kill 10 rats" achievement to be completed first.
+```python
+ACHIEVE_TEN_RATS = {
+    "name": "The Usual",
+    "desc": "Why do all these inns have rat problems?",
+    "category": "defeat",
+    "tracking": "rat",
+    "count": 10,
+}
+
+ACHIEVE_DIRE_RATS = {
+    "name": "Once More, But Bigger",
+    "desc": "Somehow, normal rats just aren't enough any more.",
+    "category": "defeat",
+    "tracking": "dire rat",
+    "count": 10,
+    "prereqs": "ACHIEVE_TEN_RATS",
+}
+```
+
+An achievement for buying 5 each of apples, oranges, and pears.
+```python
+FRUIT_BASKET_ACHIEVEMENT = {
+    "name": "A Fan of Fruit", # note, there is no desc here - that's allowed!
+    "category": "buy",
+    "tracking": ("apple", "orange", "pear"),
+    "count": 5,
+    "tracking_type": "separate",
+}
```

Comment:
After some thought: I can appreciate the usefulness of this as an approach, but I've personally found it to be more confusing to actually use in practice, personally, and it would add an extra layer of complexity to the install/setup directions, so I've opted not to implement it.

Reply:
Ok. I find this style to be _considerably_ more useful in practice, but I guess it's a matter of preference. :smiley: